draft-ietf-httpapi-ratelimit-headers.md

```diff
@@ -1138,6 +1160,17 @@ RateLimit-Limit: 100, 100;w=60;burst=1000;comment="sliding window", 5000;w=3600;
     or when they are an active component of the service.
     In those case we will consider them as part of the originating infrastructure.
 
+14. Why the `window` parameter is just informative?
+    Could it be used by a client to determine the request rate?
+
+    A non-informative `window` parameter might be fine in an environment
+    where client and servers are thighly coupled. Conveying policies
+    with this detail on a large scale would be very complex and implementations
+    would be likely not interoperable. We thus decided to leave `w` as 
+    an informational parameter and only rely on `RateLimit-Limit`,
+    `RateLimit-Remaining` and `RateLimit-Reset` for defining the throttling
+    behavior.
```

Comment:
In connection with the above changes, we might want to have another informational parameter `r` to convey an expected _maximum rate_ in quota units.

The problem might then arise from not really knowing how many units requests consume until you have performed a few of them. Providing a hint based on the current request might also require another parameter, say `u`. The following example might convey the rate to be 25 requests like this one in terms of quota-unit cost per second:

```
~~~ example
RateLimit-Limit: 10000, 10000;w=1000,r=250,u=10
RateLimit-Remaining: 10000
RateLimit-Reset: 10
~~~
```

The benefit of doing this is that the client might have better information about the effective maximum rate before performing more requests and actually forcing the service to modify the returned values.

The bad thing is that we make it a bit more complex and add more standardized informational (but optional) parameters when we have tried to leave this area more or less pristine to allow future drafts to expand on.

Reply:
Could exposing a maximum rate make things more complex without gains?

1. it is hard to enforce since it is conveyed in the informative part of the fields;
2. clients might be tempted to consider that value "the real limit";

Since parameters are always allowed in `Limit` we could define a "registry" and add them subsequently as soon as implementers needs them.

Reply:
Please see the latest proposal https://github.com/ietf-wg-httpapi/ratelimit-headers/pull/34/files#diff-d26a6f407efe0e4913cbcf7bf0e2a164bee5edea8bf2fdc4aa97a7ceb3acc03cR950

Reply:
More complex for sure, but more information is conveyed at once, and the server can always fall back to changing the values to force clients who don't understand these informational parameters to lower their rates.

Agree on having some place or section in the document to track these parameters, even if we just populate it with `w`, so any extensions can refer to and expand to it.

Reply:
@unleashed filed #49 Limit parameter registry